kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableImpl.java

```diff
@@ -200,11 +206,15 @@ public CloseableIterator<ColumnarBatch> getChanges(
                   TableFeatures.validateKernelCanReadTheTable(protocol, getDataPath().toString());
                 }
               }
+              ColumnarBatch batchToReturn = batch;
               if (shouldDropProtocolColumn) {
-                return batch.withDeletedColumnAt(protocolIdx);
-              } else {
-                return batch;
+                batchToReturn = batchToReturn.withDeletedColumnAt(protocolIdx);
               }
+              int commitInfoIdx = batchToReturn.getSchema().indexOf("commitInfo");
```

Comment:
nit - put this by `protocolIdx` for consistency?

Reply:
Gotcha. This will be changed/fixed in the future anyways when we revisit our expression support so fine by me for now.